extension/src/plots/model/collect.ts

```diff
@@ -376,6 +477,80 @@ export const collectSelectedComparisonPlots = ({
   return acc
 }
 
+const collectSelectedImgComparisonPlotClasses = ({
+  comparisonClassesSelected,
+  acc,
+  img,
+  id,
+  path
+}: {
+  comparisonClassesSelected: ComparisonClassesSelected
+  acc: ComparisonPlotClasses
+  img: ImagePlot
+  id: string
+  path: string
+}) => {
+  const imgClasses: ComparisonPlotClass[] = []
+
+  for (const [label, boxes] of Object.entries(img.boundingBoxes || {})) {
+    const selectedState = comparisonClassesSelected[path]?.[label]
+
+    if (selectedState === false) {
+      continue
+    }
+
+    imgClasses.push({
+      boxes: boxes.map(({ right, left, top, bottom }) => ({
+        h: bottom - top,
+        w: right - left,
+        x: left,
+        y: top
+      })),
+      label
+    })
+  }
+
+  if (imgClasses.length === 0) {
+    return
+  }
+
+  if (!acc[id]) {
+    acc[id] = {}
+  }
+
+  acc[id][path] = imgClasses
+}
+
+export const collectSelectedComparisonPlotClasses = ({
+  comparisonClassesSelected,
+  comparisonData,
+  paths,
+  selectedRevisionIds
+}: {
+  comparisonClassesSelected: ComparisonClassesSelected
+  comparisonData: ComparisonData
+  paths: string[]
+  selectedRevisionIds: string[]
+}) => {
+  const acc: ComparisonPlotClasses = {}
+
+  for (const path of paths) {
+    for (const id of selectedRevisionIds) {
+      for (const img of comparisonData[id][path]) {
+        collectSelectedImgComparisonPlotClasses({
```

Comment:
Is there a way to refactor `collectSelectedImgComparisonPlotClasses` so that it does not have to be nested inside three loops?

Reply:
Updated the code to no longer have three nested loops!